### PR TITLE
Checkpoint Harmony connector

### PR DESCRIPTION
**Pull request by vg-svitla**

### CODE REVIEW

File: Checkpoint/connectors/checkpoint_harmony.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        while self.running:
+            try:
+                loop = asyncio.get_event_loop()
+
+                while self.running:
+                    processing_start = time.time()
+                    if previous_processing_end is not None:
+                        EVENTS_LAG.labels(intake_key=self.configuration.intake_key).observe(
+                            processing_start - previous_processing_end
+                        )
```
Comment (squioc) [inline Checkpoint/connectors/checkpoint_harmony.py:129]: Can you use the `latest_event_date` to compute the lag? Like in the [cybereason connector](https://github.com/SEKOIA-IO/automation-library/blob/670123d1651c506afbf8ae8ef6a214c44aa33151/Cybereason/cybereason_modules/connector_pull_events.py#L316-L322)?

File: Checkpoint/connectors/metrics.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+FORWARD_EVENTS_DURATION = Histogram(
+    name="forward_events_duration",
+    documentation="Duration to collect and forward events from eventhub",
+    namespace=prom_namespace,
+    labelnames=["intake_key"],
+)
+
+EVENTS_LAG = Histogram(
+    name="event_lags",
```
Comment (squioc) [inline Checkpoint/connectors/metrics.py:30]:
```suggestion
    name="events_lags",
```

File: Checkpoint/connectors/checkpoint_harmony.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            self._checkpoint_client = CheckpointHttpClient(
+                client_id=self.module.configuration.client_id,
+                secret_key=self.module.configuration.secret_key,
+                auth_url=self.module.configuration.authentication_url,
+                base_url=self.module.configuration.base_url,
+            )
+
+        return self._checkpoint_client
+
+    async def get_checkpoint_harmony_events(self) -> list[str]:
```
Comment (squioc) [inline Checkpoint/connectors/checkpoint_harmony.py:81]: How does this method handle pages of events when interacting with the API? I can't find how the client iterates over the pages in this method nor in the `get_harmony_mobile_alerts` method.

File: Checkpoint/manifest.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            "secret_key"
+        ],
+        "title": "Check Point module Configuration",
+        "type": "object"
+    },
+    "description": "Check Point Harmony is the industry’s first unified security solution for users devices and access.",
+    "name": "Check Point",
+    "uuid": "096f4eda-68dd-11ee-8c99-0242ac120002",
+    "slug": "checkpoint",
+    "version": "1.0.2"
```
Comment (squioc) [inline Checkpoint/manifest.json:38]:
```suggestion
    "version": "1.0.0"
```

**Review by squioc — CHANGES_REQUESTED** [review]
Thanks for the job.
Some changes to apply and some questions.

File: Checkpoint/connectors/checkpoint_harmony.py (hunk 3 above)
Comment (vg-svitla) [reply Checkpoint/connectors/checkpoint_harmony.py:81]: Post an update in DM

**Review by squioc — APPROVED** [review]

**Pull request merged**